Source/AssetRipper.Processing/AnimatorControllers/AnimatorStateMachineContext.cs

```diff
@@ -875,13 +874,13 @@ private void ProcessChildrenPosForStateMachine(IAnimatorStateMachine stateMachin
 				// Position all Child StateMachines second 
 				else if (stateMachine.Has_ChildStateMachines())
 				{
-					// remember to handle Unity 5- SubStateMachines too
 					ChildAnimatorStateMachine csm = stateMachine.ChildStateMachines[i - stateCount];
 					csm.Position.CopyValues(position);
 				}
 				else
 				{
-					stateMachine.ChildStateMachinePosition.Add(position);
+					Vector3f newPos = stateMachine.ChildStateMachinePosition.AddNew();
+					newPos.CopyValues(position);
```

Comment:
This can be one line. Also, `position` should be changed to `System.Numerics.Vector3`. I didn't notice that in your first pull request.